src/Middleware/Diagnostics.EntityFrameworkCore/src/MigrationsEndPointMiddleware.cs

```diff
@@ -15,6 +15,7 @@ namespace Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore
     /// <summary>
     /// Processes requests to execute migrations operations. The middleware will listen for requests to the path configured in the supplied options.
     /// </summary>
+    [Obsolete("This is obsolete and will be removed in a future version. Use the Package Manager Console in Visual Studio or dotnet-ef tool on the command line to apply migrations.", error:true)]
     public class MigrationsEndPointMiddleware
```

Comment:
This is dead code now. Remove all internal content and make the public APIs throw NotImplementedException.